docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,388 @@
+Running `kubeadm init federation` first performs all the operations
+involved in bootstrapping a Kubernetes control plane, as running
+`kubeadm init` would. We henceforth call this Kubernetes cluster for
+which the control plane was initialized, the "bootstrap cluster". Upon
+initializing the bootstrap cluster, we perform the following operations
+to bootstrap a federation control plane.
+
+
+**1. Create a namespace for federation system components**
+
+* Call this namespace `federation-system` to keep it consistent with
+  Kubernetes.
+* The namespace is created in the bootstrap cluster.
+
+**2. Expose a network endpoint for the federation API server**
+
+* Create a "Loadbalancer" type Kubernetes service in the
+  `federation-system` namespace in the bootstrap cluster to expose the
+  yet to be created federation API server.
+* Wait until a load balancer IP address is allocated to the service.
+
+**3. Generate TLS certificates and credentials**
+
+* Using `kubeadm`'s PKI infrastructure, generate TLS certificates for a
+  new CA, yet to be created federation API server and federation
+  controller manager.
+* Generate credentials for the federation API server.
+
+**4. Create a kubeconfig secret**
+
+* Using the load balancer IP address of the federation API server and
+  the generated certificates and credentials, generate a kubeconfig
+  file for the federation controller manager.
+* Create a secret in the bootstrap cluster's `federation-system`
+  namespace and populate it with the contents of the generated
+  kubeconfig file.
+
+**5. Create a persistent volume and a claim to store the federation API server's state**
+
+* Create a persistent volume claim in the `federation-system` namespace.
+* Also request a dynamically provisioned persistent storage device for
+  the PVC.
+
+**6. Create federation API server**
+
+* Create a deployment for the federation API server in the `federation-
+  system` namespace.
+* The pod template spec is composed of two containers: one for `etcd`
+  and the other one for the federation API server.
+* The pod template spec references the PVC created in the previous step
+  as a volume and the `etcd` container mounts that volume to store
+  `etcd`'s data.
+* The federation API server container uses the official `hyperkube`
+  release image.
+* Since the API server is stateless, it needs no persistent storage.
+  However, the certificates and the credentials are made available to
+  the API server by mounting them as volumes inside the API server
+  container.
+
+**7. Create federation controller manager**
+
+* Create a deployment for the federation controller manager in the
+  `federation-system` namespace.
+* The pod template spec consists of a single container for the
+  federation controller manager and it also uses the same official
+  `hyperkube` release image as the federation API server.
+* Kubeconfig required by the federation controller manager to discover
```

Comment:
Regarding discovery, I think we can use DNS (of the known API service name)?  Just a thought.  Might be simpler.